src/metabase/lib/field.cljc

```diff
@@ -96,6 +96,18 @@
                     {::original-effective-type original-effective-type})
                   (when-let [original-temporal-unit (::original-temporal-unit opts)]
                     {::original-temporal-unit original-temporal-unit})
+                  ;; `:inherited-temporal-unit` is transfered from `:temoral-unit` ref option only when
+                  ;; the [[lib.metadata.calculation/*propagate-inherited-temoral-unit*]] is thruthy, ie. bound. Intent
```

Comment:
I wonder if using dynamic vars is the best way of doing this. Ideally the transition of `:temporal-unit` to `:inherited-temporal-unit` where we jump to the next stage.